frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideInBackgroundable.java

```diff
@@ -0,0 +1,158 @@
+import com.virtuslab.branchlayout.api.EntryIsDescendantOfException;
+import com.virtuslab.branchlayout.api.EntryIsRootException;
+import com.virtuslab.branchlayout.api.IBranchLayout;
+import com.virtuslab.branchlayout.api.IBranchLayoutEntry;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.SlideInOptions;
+
+public class SlideInBackgroundable extends Task.Backgroundable {
+
+  private final GitRepository gitRepository;
+  private final IBranchLayout branchLayout;
+  private final IBranchLayoutWriter branchLayoutWriter;
+  private final Runnable preSlideInRunnable;
+  private final SlideInOptions slideInOptions;
+  private final String parentName;
+  private final VcsNotifier notifier;
+
+  public SlideInBackgroundable(
+      Project project,
+      GitRepository gitRepository,
+      IBranchLayout branchLayout,
+      IBranchLayoutWriter branchLayoutWriter,
+      Runnable preSlideInRunnable,
+      SlideInOptions slideInOptions,
+      String parentName) {
+    super(project, getString("action.GitMachete.BaseSlideInBranchBelowAction.task-title"));
+    this.gitRepository = gitRepository;
+    this.branchLayout = branchLayout;
+    this.branchLayoutWriter = branchLayoutWriter;
+    this.preSlideInRunnable = preSlideInRunnable;
+    this.slideInOptions = slideInOptions;
+    this.parentName = parentName;
+    this.notifier = VcsNotifier.getInstance(project);
+  }
+
+  @Override
+  public void run(ProgressIndicator indicator) {
+    preSlideInRunnable.run();
+
+    // `preSlideInRunnable` may perform some sneakily-asynchronous operations (e.g. checkoutRemoteBranch).
+    // The high-level method used within the runnable do not allow us to schedule the tasks after them.
+    // (Stepping deeper is not an option since we loose some important logic or make us very dependent on git4idea.)
+    // Hence we await for the creation of the branch (with exponential backoff).
+    waitForLocalBranch();
+
+    Path macheteFilePath = getMacheteFilePath(gitRepository);
+
+    var childEntryByName = branchLayout.findEntryByName(slideInOptions.getName());
+    IBranchLayoutEntry entryToSlideIn;
+    IBranchLayout targetBranchLayout;
+    if (childEntryByName.isDefined()) {
+      var entryByName = childEntryByName.get();
+
+      if (slideInOptions.shouldReattach()) {
+        entryToSlideIn = entryByName;
+        targetBranchLayout = branchLayout;
+      } else {
+        entryToSlideIn = entryByName.withChildren(List.empty());
+        targetBranchLayout = Try.of(() -> branchLayout.slideOut(slideInOptions.getName()))
+            .onFailure(EntryDoesNotExistException.class, exceptionWithMessageHandler(
+                format(getString("action.GitMachete.BaseSlideInBranchBelowAction.notification.message.entry-does-not-exist"),
+                    entryToSlideIn.getName()))::apply)
+            .onFailure(EntryIsRootException.class, exceptionWithMessageHandler(
```

Comment:
Maybe `exceptionWithMessageHandler` could return a ready `Consumer`? Then `::apply` won't be necessary